contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -214,6 +263,34 @@ contract LidoOracle is ILidoOracle, AragonApp {
         _tryPush(_epochId);
     }
 
+    /**
+     * @notice To make oracles less dangerous,
+     *  we can bound rewards report by 0.1% increase in stake and 15% decrease in stake,
+     *  with both values configurable by DAO voting in case of extremely unusual circumstances.
+     *  daily_reward_rate_PPM = 1e6 * reward / totalPooledEther / days
+     **/
+    function reportSanityChecks(Report report) private {
+        (uint256 lastTotalPooledEther,
+         uint256 curTotalPooledEther,
+         uint256 lastValidators,
+         uint256 timeElapsed) = getLastCompletedReport();
+        // todo: use safeMath everywhere
+        if (report.curTotalPooledEther >= lastTotalPooledEther) {  // check profit constraint
+            uint256 reward = curTotalPooledEther - lastTotalPooledEther;
+            uint256 rewardPPM = 1e6 * reward / curTotalPooledEther * DAY / timeElapsed;
+            require(rewardPPM <= ALLOWED_BEACON_BALANCE_INCREASE, "ALLOWED_BEACON_BALANCE_INCREASE");
+        } else {  // check loss constraint (???)
+            uint256 loss = lastTotalPooledEther - curTotalPooledEther;
+            uint256 lossPPM = 1e6 * loss / curTotalPooledEther * DAY / timeElapsed;
+            require(lossPPM <= ALLOWED_BEACON_BALANCE_DECREASE, "ALLOWED_BEACON_BALANCE_DECREASE");
+        }
+
+        if (report.beaconValidators >= lastValidators) {
```

Comment:
I think we do not want to limit the growth of validators. This is from our latest discussion in chat, not from the issue.